homepage/homepage_web_client.go

```diff
@@ -0,0 +1,53 @@
+package homepage
+
+import (
+	"context"
+	"fmt"
+	"github.com/ONSdigital/dp-frontend-homepage-controller/dpcache"
+	"time"
+)
+
+type HomepageWebClient struct {
+	HomepageUpdater
+	cache     dpcache.DpCacher
+	languages []string
+}
+
+func NewHomePageWebClient(clients *Clients, updateInterval time.Duration, languages []string) HomepageClienter {
+	return &HomepageWebClient{
+		HomepageUpdater: HomepageUpdater{
+			clients: clients,
+		},
+		cache:     dpcache.NewDpCache(updateInterval),
+		languages: languages,
+	}
+}
+
+func (hwc *HomepageWebClient) GetHomePage(ctx context.Context, userAccessToken, collectionID, lang string) (string, error) {
+	homePageCachedHTML, ok := hwc.cache.Get(getCachingKeyForLanguage(lang))
+	if ok {
+		homePageString, convertible := homePageCachedHTML.([]byte)
+		if convertible {
+			return string(homePageString), nil
+		}
+	}
+
+	return "", fmt.Errorf("failed to read homepage from cache")
+
+}
+
+func getCachingKeyForLanguage(lang string) string {
+	return fmt.Sprintf("%s___%s", dpcache.HomepageCacheKey, lang)
+}
+
+func (hwc *HomepageWebClient) StartBackgroundUpdate(ctx context.Context, errorChannel chan error) {
+	for _, lang := range hwc.languages {
+		langKey := getCachingKeyForLanguage(lang)
+		hwc.cache.AddUpdateFunc(langKey, hwc.GetHomePageUpdateFor(ctx, "", "", lang))
```

Comment:
Just a question to see if I fully understand the code: `StartBackgroundUpdate` registers the `GetHomePageUpdateFor` function so that we can get the page content when we don't have a cache. It will only be called in web mode and that's why we don't provide a token.

Will this function be used only in web mode? (otherwise, it might be worth allowing tokens to be passed to it?) - This is just a question, not a change request, if it's intended to be used only in web mode then that is ok.

Reply:
Yes this is used only in the web mode.